spec/rails_helper.rb

```diff
@@ -119,108 +97,49 @@ def stub_addresses
+
+    # Stub out the Geocoder since we
+    Geocoder.configure(lookup: :test)
+
+    ["1500 Remount Road, Front Royal, VA 22630",
+     "123 Donation Site Way",
+     "Smithsonian Conservation Center new"].each do |address|
+       Geocoder::Lookup::Test.add_stub(
+         address, [
+           {
+             "latitude" => 40.7143528,
+             "longitude" => -74.0059731,
+             "address" => "1500 Remount Road, Front Royal, VA",
+             "state" => "Virginia",
+             "state_code" => "VA",
+             "country" => "United States",
+             "country_code" => "US"
+           }
+         ]
+       )
+     end
+
+    seed_base_items_for_tests
   end
 
   config.before(:each, type: :system) do
-    # Use truncation in the case of doing `browser` tests because it
-    # appears that transactions won't work since it really does
-    # depend on the database to have records.
-    DatabaseCleaner.strategy = :truncation
-  end
-
-  config.before(:each, type: :request) do
-    # Use truncation in the case of doing `browser` tests because it
-    # appears that transactions won't work since it really does
-    # depend on the database to have records.
-    DatabaseCleaner.strategy = :truncation
-  end
-
-  config.before(:each) do |example|
-    if example.metadata[:persisted_data]
-      DatabaseCleaner.strategy = :truncation
-    else
-      # The database cleaner will now begin at this point
-      # up anything after this point when `.clean` is called.
-      DatabaseCleaner.start
-
-      # "Dirty" the database by adding the essential records
-      # necessary to run tests.
-      #
-      # If you are using :transaction, it will just rollback any additions
-      # when `.clean` is called. Any previous changes will be kept prior to
-      # the call `DatabaseCleaner.start`
-      #
-      # If you are using :truncation, it will erase everything once `.clean`
-      # is called.
-      seed_base_items_for_tests
-      # Always create organization, almost no tests can be run without one
-      create_organization
-
-      seed_with_default_records unless example.metadata[:skip_seed]
-    end
-  end
-
-  config.before(:each, type: proc { |v| %i[request system controller].include?(v) }) do
-    create_default_users
-  end
-
-  config.before(:each, :needs_users) do
-    create_default_users
+    clear_downloads
+    driven_by :chrome
+    Capybara.server = :puma, { Silent: true }
   end
 
-  config.after(:each) do |example|
-    # Ensure to clean-up the database by whichever means
-    # were specified before the test ran
-    DatabaseCleaner.clean unless example.metadata[:persisted_data]
-
-    # Remove any /tmp/storage files that might have been
-    # added as a consequence of the test.
-    FileUtils.rm_rf(Dir["#{Rails.root}/tmp/storage"])
+  config.before(:each) do
+    # Seeding users in the `each` block because it makes the @ variables defined
+    # through the specs
+    seed_users
```

Comment:
Can we seed users and items at the same time and then just use `select` queries to fetch the created items? It's way faster to read than write on every single spec.

Reply:
@dorner good idea!